build_functions/CMakeLists.txt

```diff
@@ -109,6 +112,67 @@ function(build_lib whatIsBuilding solution_folder)
         target_sources(${whatIsBuilding}_lib_${CMAKE_PROJECT_NAME} PRIVATE ${MOCKS_SOURCE_FILE})
     endif()
 
+    # ENABLE_TEST_FILES_PRECOMPILED_HEADERS creates precompiled headers for the test file
+    # It expects that exactly one test file exists. This is because cmake's target_precompile_headers
+    # only allows one precompiled headers file.
+    # If a specific header filename is provided, that is used.
+    # If just the flag is provided, the header filename is by convention computed from the test file by replacing the extension with .h
+    # A _mocks.c file is generated that includes the precompiled header without ENABLE_MOCKS_DECL,
+    # having the purpose of providing the mock implementations.
+    if(arg_ENABLE_TEST_FILES_PRECOMPILED_HEADERS)
+        # Check that exactly one test file exists (required for all cases)
+        list(LENGTH ${whatIsBuilding}_test_files TEST_FILES_COUNT)
+        if(NOT TEST_FILES_COUNT EQUAL 1)
+            message(WARNING "ENABLE_TEST_FILES_PRECOMPILED_HEADERS requires exactly one test file, but found ${TEST_FILES_COUNT} files in ${whatIsBuilding}_test_files: ${${whatIsBuilding}_test_files}")
+        else()
+            # Check if a specific header file was provided
+            if(NOT (arg_ENABLE_TEST_FILES_PRECOMPILED_HEADERS STREQUAL "ON" OR arg_ENABLE_TEST_FILES_PRECOMPILED_HEADERS STREQUAL "TRUE"))
+                # A specific header filename was provided
+                set(PRECOMPILED_HEADER_FILENAME "${arg_ENABLE_TEST_FILES_PRECOMPILED_HEADERS}")
+            else()
+                # No specific header provided, compute it from the first test file
+                # Get the single test file
+                list(GET ${whatIsBuilding}_test_files 0 TEST_FILE)
+                
+                # Compute the header filename by replacing extension with .h
+                get_filename_component(TEST_FILE_NAME_WE ${TEST_FILE} NAME_WE)
```

Comment:
>WE [](http://example.com/codeflow?start=75&length=2)

WE as in First Person Plural Noun or WE and in Without Extension?

I might guess it is WithoutExtension, but a clear spelling would be even nicer in 2 years when we revisit this code.

Reply:
Fun fact is this is cmake naming:
[https://cmake.org/cmake/help/latest/command/get_filename_component.html#command:get_filename_component](https://cmake.org/cmake/help/latest/command/get_filename_component.html#command:get_filename_component)

So the mode I cannot change.
The bot changed the var name to spell WITHOUT_EXTENSION.